### ISSUE TITLE
Live stream Widevine decryption question

### ISSUE THREAD

**Issue by peyoh** [google/shaka-player#309]
Hello,
I'm trying with no success to play Widevine encrypted live stream.
For packetizer I'm using edash-packager. 
Can you help me what may be wrong? 

Chrome version is 46.0.2490.71 (64-bit), Linux.

 chrome://media-internals log shows:
```
00:00:00 00	pipeline_state	kCreated
00:00:00 00	event	WEBMEDIAPLAYER_CREATED
00:00:00 00	url	blob:http%3A//213.167.17.218%3A81/4c081c0c-7e04-4c12-be10-660c15655912
00:00:00 00	pipeline_state	kInitDemuxer
00:00:00 07	duration	2673719.558333
00:00:00 18	info	Audio codec: mp4a.40.2
00:00:00 18	info	Audio codec: mp4a.40.2
00:00:00 18	found_audio_stream	true
00:00:00 18	audio_codec_name	aac
00:00:00 20	info	Video codec: avc1.4d015
00:00:00 20	info	Video codec: avc1.4d015
00:00:00 20	found_video_stream	true
00:00:00 20	video_codec_name	h264
00:00:00 20	pipeline_state	kInitRenderer
00:00:00 26	debug	DecryptingAudioDecoder: failed to init decoder on decryptor
00:00:00 27	audio_dds	true
00:00:00 27	audio_decoder	FFmpegAudioDecoder
00:00:00 28	debug	Video rendering in low delay mode.
00:00:00 33	video_dds	false
00:00:00 33	video_decoder	DecryptingVideoDecoder
00:00:00 33	pipeline_state	kPlaying
00:00:00 33	seek_target	81719.5546875
00:00:00 33	pipeline_state	kSeeking
00:00:00 122	error	Incorrect CENC subsample size.
00:00:00 122	error	Append: stream parsing failed. Data size=131072 append_window_start=0 append_window_end=inf
00:00:00 122	pipeline_error	pipeline: decode error
00:00:00 122	pipeline_state	kStopping
00:00:00 123	pipeline_state	kStopped
```

In the browser's console:
```

+ audio/mp4; codecs="mp4a.40.2" is supported
player.js:191 + video/mp4; codecs="avc1.4d0015" is supported
player.js:191 + video/mp4; codecs="avc1.4d001e" is supported
player.js:191 + video/mp4; codecs="avc1.4d0015" is supported
player.js:191 + video/mp4; codecs="avc1.4d001e" is supported
eme_manager.js:291 MKSC Object {audioCapabilities: Array[1], videoCapabilities: undefined, initDataTypes: undefined, distinctiveIdentifier: "optional", persistentState: "optional"…}
eme_manager.js:291 MKSC Object {audioCapabilities: Array[1], videoCapabilities: Array[1], initDataTypes: undefined, distinctiveIdentifier: "optional", persistentState: "optional"…}
live_segment_index.js:240 @availabilityStartTime seems to be inaccurate; some segments may not be available yet: currentPresentationTime 81718.0039999485 latestAvailableSegmentEndTime 81731.53866666666
live_segment_index.js:240 @availabilityStartTime seems to be inaccurate; some segments may not be available yet: currentPresentationTime 81718.0039999485 latestAvailableSegmentEndTime 81736.28
stream_video_source.js:1568 Starting each stream from 81719.55833342871
eme_manager.js:546 onEncrypted_ [0, 0, 0, 116, 112, 115, 115, 104, 0, 0, 0, 0, 237, 239, 139, 169, 121, 214, 74, 206, 163, 200, 39, 220, 213, 29, 33, 237, 0, 0, 0, 84, 8, 1, 18, 16, 8, 245, 171, 99, 112, 217, 88, 77, 181, 86, 215, 169, 170, 113, 115, 141, 18, 16, 125, 24, 221, 73, 168, 111, 86, 39, 176, 161, 80, 236, 131, 157, 161, 82, 18, 16, 86, 27, 149, 159, 72, 126, 87, 142, 169, 108, 51, 55, 246, 148, 164, 2, 26, 12, 105, 110, 118, 101, 110, 116, 111, 115, 115, 121…] MediaEncryptedEvent {}
eme_manager.js:662 onSessionMessage_ MediaKeyMessageEvent {}
app.js:884 Player error CustomEvent {isTrusted: false, srcElement: null, currentTarget: s…a.p…r.Player, target: s…a.m…a.Stream}app.onPlayerError_ @ app.js:884shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:167shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:180shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:180shaka.util.FakeEventTarget.dispatchEvent @ fake_event_target.js:145(anonymous function) @ stream.js:658
eme_manager.js:546 onEncrypted_ [0, 0, 0, 116, 112, 115, 115, 104, 0, 0, 0, 0, 237, 239, 139, 169, 121, 214, 74, 206, 163, 200, 39, 220, 213, 29, 33, 237, 0, 0, 0, 84, 8, 1, 18, 16, 8, 245, 171, 99, 112, 217, 88, 77, 181, 86, 215, 169, 170, 113, 115, 141, 18, 16, 125, 24, 221, 73, 168, 111, 86, 39, 176, 161, 80, 236, 131, 157, 161, 82, 18, 16, 86, 27, 149, 159, 72, 126, 87, 142, 169, 108, 51, 55, 246, 148, 164, 2, 26, 12, 105, 110, 118, 101, 110, 116, 111, 115, 115, 121…] MediaEncryptedEvent {}
app.js:884 Player error CustomEvent {isTrusted: false, srcElement: null, currentTarget: s…a.p…r.Player, target: s…a.p…r.Player}app.onPlayerError_ @ app.js:884shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:167shaka.util.FakeEventTarget.dispatchEvent @ fake_event_target.js:145shaka.player.Player.onError_ @ player.js:433
app.js:884 Player error CustomEvent {isTrusted: false, srcElement: null, currentTarget: s…a.p…r.Player, target: s…a.m…a.Stream}app.onPlayerError_ @ app.js:884shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:167shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:180shaka.util.FakeEventTarget.recursiveDispatch_ @ fake_event_target.js:180shaka.util.FakeEventTarget.dispatchEvent @ fake_event_target.js:145(anonymous function) @ stream.js:658
eme_manager.js:770 onLicenseSuccess_ MediaKeySession {}
eme_manager.js:778 onSessionReady_ MediaKeySession {}
```

**Comment by joeyparrish** [comment]
Just looking at your media-internals log, it seems that this is the problem from Chrome's perspective:

`00:00:00 122    error   Incorrect CENC subsample size.`

This message comes from Chrome's media stack:

https://code.google.com/p/chromium/codesearch#chromium/src/media/formats/mp4/track_run_iterator.cc&l=610

This is outside of Shaka Player's control, so you'll need to file this bug against either the packager or Chrome itself.

**Comment by kqyang** [comment]
I believe you are using packager code @ master, between 96abd90ca29bda736e0ddb3ec1bd34aab977c559 and ef81be5f7b5c24fdda779cee272bb45777b7800e. There is a known problem, which has been fixed in 1863f5a569340132760ac3081c70a249ed72ff14.

Please try with 1863f5a569340132760ac3081c70a249ed72ff14 or later. Sorry about the inconvenience.
